static/js/operation.js

```diff
@@ -165,89 +127,69 @@ function parseAssembly(text, data) {
   var length = 0;
   var gc = 0;
   var coverage = 0;
-  var contig_length = 0;
-  var GC_count = 0;
+  var currentContigLength = 0;
+  var gcCount = 0;
 
   for (var i = 0; i < lines.length; i++) {
     var line = lines[i];
-    if (line.toUpperCase().match(/^[ATGCU]+$/g)) { // Checking if the current line is not a contig title
-      [GC_count, contig_length] = calculateGCAndLength(line, GC_count, contig_length);
-      gc = ((GC_count/contig_length).toFixed(2)).toString(); // idnetified last component of contig title
+    if (line.search(/^>{1}/gm) === 0) { // Checking if the current line is a contig title
+        [id, length, coverage] = parseContigTitles(line, format);
+        if (length <= minContigLength) 
+          continue;
+
+        // append dataframe with contig title information
+        df.push([id, length, , coverage]) // gc not calculated yet
+        gcCount = 0;
+        currentContigLength = 0;
     }
     else {
-      if (i !== 0) {
-        if (length <= 1000) {
-          continue;
-        }
-        df.push([id, length, gc, coverage]) // append dataframe with contig title information
-        GC_count = 0;
-        contig_length = 0;
+      gcCount += calculateLineGC(line);
+      currentContigLength += line.length; 
+      if (df.length != 0 && currentContigLength === Number(length)) { // verifying if current line is last line in contig
```

Comment:
This verification may work but it depends on the correctness of the "length" field in the title. In the future, there may be scenarios where the title only contains coverage or not containing anything. In that case, the length needs to be found out from the sequence itself.

A convenient way is that: Do this check at the beginning of the next contig. That is, when the first character of a line is `>`. Then, in the end after reading the entire file (after the for loop), do this again.
```